libsolidity/inlineasm/AsmData.h

```diff
@@ -38,45 +38,202 @@ namespace assembly
+	std::string name{};
+};
+
 /// Assignment from stack (":= x", moves stack top into x, potentially multiple slots)
-struct StackAssignment { SourceLocation location; Identifier variableName; };
+struct StackAssignment
+{
+	StackAssignment() {}
+	StackAssignment(SourceLocation const& location, Identifier const& variableName):
+			location(location), variableName(variableName) {}
+	SourceLocation location{};
+	Identifier variableName{};
+};
+
 /// Assignment ("x := mload(20:u256)", expects push-1-expression on the right hand
 /// side and requires x to occupy exactly one stack slot.
 ///
 /// Multiple assignment ("x, y := f()"), where the left hand side variables each occupy
 /// a single stack slot and expects a single expression on the right hand returning
 /// the same amount of items as the number of variables.
-struct Assignment { SourceLocation location; std::vector<Identifier> variableNames; std::shared_ptr<Expression> value; };
+struct Assignment
+{
+	Assignment() {}
+	Assignment(SourceLocation const& location, std::vector<Identifier> const& variableNames,
+			   std::shared_ptr<Expression> const& value):
+			location(location), variableNames(variableNames), value(value) {}
+	SourceLocation location{};
+	std::vector<Identifier> variableNames{};
+	std::shared_ptr<Expression> value{};
+};
+
 /// Functional instruction, e.g. "mul(mload(20:u256), add(2:u256, x))"
-struct FunctionalInstruction { SourceLocation location; solidity::Instruction instruction; std::vector<Expression> arguments; };
-struct FunctionCall { SourceLocation location; Identifier functionName; std::vector<Expression> arguments; };
+struct FunctionalInstruction
+{
+	FunctionalInstruction() {}
+	FunctionalInstruction(SourceLocation const& location, solidity::Instruction instruction,
+						  std::vector<Expression> const& arguments):
+			location(location), instruction(instruction), arguments(arguments) {}
+	SourceLocation location{};
+	solidity::Instruction instruction{};
+	std::vector<Expression> arguments{};
+};
+
+struct FunctionCall
+{
+	FunctionCall() {}
+	FunctionCall(SourceLocation const& location, Identifier const&functionName,
+				 std::vector<Expression> const& arguments):
+			location(location), functionName(functionName), arguments(arguments) {}
+	SourceLocation location{};
+	Identifier functionName{};
+	std::vector<Expression> arguments{};
+};
+
 /// Statement that contains only a single expression
-struct ExpressionStatement { SourceLocation location; Expression expression; };
+struct ExpressionStatement
+{
+	ExpressionStatement() {}
+	ExpressionStatement(SourceLocation const& location, Expression const& expression):
+			location(location), expression(expression) {}
+	SourceLocation location{};
+	Expression expression{};
+};
+
 /// Block-scope variable declaration ("let x:u256 := mload(20:u256)"), non-hoisted
-struct VariableDeclaration { SourceLocation location; TypedNameList variables; std::shared_ptr<Expression> value; };
+struct VariableDeclaration
```

Comment:
I don't think this is where we want to go...

Reply:
I guess there are some optimzations possible ;) At first I wanted to get rid of all the warnings..